ceremonies/src/tests.rs

```diff
@@ -1124,126 +1216,377 @@ fn grow_population_works() {
-	}
+		let participants: Vec<AccountId> =
+			add_population(12, 0).iter().map(|b| account_id(&b)).collect();
 
-	participants
+		BootstrapperRegistry::<TestRuntime>::insert((cid, cindex), 1, participants[0].clone());
+		BootstrapperRegistry::<TestRuntime>::insert((cid, cindex), 2, participants[1].clone());
+		BootstrapperRegistry::<TestRuntime>::insert((cid, cindex), 3, participants[2].clone());
+
+		ReputableRegistry::<TestRuntime>::insert((cid, cindex), 1, participants[3].clone());
+		ReputableRegistry::<TestRuntime>::insert((cid, cindex), 2, participants[4].clone());
+		ReputableRegistry::<TestRuntime>::insert((cid, cindex), 3, participants[5].clone());
+
+		EndorseeRegistry::<TestRuntime>::insert((cid, cindex), 1, participants[6].clone());
+		EndorseeRegistry::<TestRuntime>::insert((cid, cindex), 2, participants[7].clone());
+		EndorseeRegistry::<TestRuntime>::insert((cid, cindex), 3, participants[8].clone());
+
+		NewbieRegistry::<TestRuntime>::insert((cid, cindex), 1, participants[9].clone());
+		NewbieRegistry::<TestRuntime>::insert((cid, cindex), 2, participants[10].clone());
+		NewbieRegistry::<TestRuntime>::insert((cid, cindex), 3, participants[11].clone());
+		AssignmentCounts::insert(
+			(cid, cindex),
+			AssignmentCount { bootstrappers: 3, reputables: 3, endorsees: 3, newbies: 3 },
+		);
+
+		MeetupCount::insert((cid, cindex), 2);
+
+		Assignments::insert(
+			(cid, cindex),
+			Assignment {
+				bootstrappers_reputables: AssignmentParams { m: 5, s1: 2, s2: 3 },
+				endorsees: AssignmentParams { m: 3, s1: 2, s2: 1 },
+				newbies: AssignmentParams { m: 3, s1: 1, s2: 2 },
+				locations: AssignmentParams { m: 9, s1: 8, s2: 7 },
+			},
+		);
+
+		let mut m0_expected_participants = [
+			participants[1].clone(),
+			participants[2].clone(),
+			participants[3].clone(),
+			participants[7].clone(),
+			participants[8].clone(),
+			participants[9].clone(),
+			participants[10].clone(),
+		];
+		let mut m1_expected_participants = [
+			participants[0].clone(),
+			participants[4].clone(),
+			participants[5].clone(),
+			participants[6].clone(),
+			participants[11].clone(),
+		];
+		let mut m0_participants = EncointerCeremonies::get_meetup_participants((cid, cindex), 1);
+		let mut m1_participants = EncointerCeremonies::get_meetup_participants((cid, cindex), 2);
+
+		m0_expected_participants.sort();
+		m1_expected_participants.sort();
+		m0_participants.sort();
+		m1_participants.sort();
+
+		assert_eq!(m0_participants, m0_expected_participants);
+		assert_eq!(m1_participants, m1_expected_participants);
+	});
+}
+
+#[rstest(
+	n_locations,
+	n_bootstrappers,
+	n_reputables,
+	n_endorsees,
+	n_newbies,
+	exp_m_bootstrappers_reputables,
+	exp_m_endorsees,
+	exp_m_newbies,
+	exp_n_assigned_bootstrappers,
+	exp_n_assigned_reputables,
+	exp_n_assigned_endorsees,
+	exp_n_assigned_newbies,
+	case(3, 7, 12, 6, 13, 19, 5, 5, 7, 12, 6, 5),
+	case(10, 1, 1, 20, 13, 2, 17, 2, 1, 1, 18, 0)
+)]
+fn generate_meetup_assignment_params_works(
```

Comment:
beautiful

Reply:
But maybe we want to put some explicit corner-cases here to be sure that they work?